src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -208,21 +208,31 @@ void CalorimeterHitReco::AlgorithmProcess() {
 //                dd4hep::Position(gpos.x(), gpos.y(), gpos.z()));//dd4hep::Position(gpos.x, gpos.y, gpos.z)
         std::vector<double> cdim;
         // get segmentation dimensions
-        if (converter->findReadout(local).segmentation().type() != "NoSegmentation") {
-            cdim = converter->cellDimensions(cellID);
+        auto segmentation_type = converter->findReadout(local).segmentation().type();
+        auto cell_dim = converter->cellDimensions(cellID);
+        if (segmentation_type == "CartesianGridXY") {
+            cdim.resize(3);
+            cdim[0] = cell_dim[0];
+            cdim[1] = cell_dim[1];
```

Comment:
Just confirming, this doesn't leave undefined `cdim[2]`?

Reply:
Yes, resize requires `T` to be `DefaultInsertable`. In order to not zero the appended items, one would need to provide a custom `Allocator`, which we, obviously, don't do.